src/js/wall/__tests__/LoginForm.test.tsx

```diff
@@ -1,74 +1,78 @@
-import { LoginForm } from "../LoginForm";
-import { useLoginMutation } from "../Queries";
-
-vi.mock("../Queries", () => ({
-    useLoginMutation: vi.fn(),
-}));
+import LoginForm from "../LoginForm";
 
 vi.mock("react-redux", () => ({
     useDispatch: vi.fn(),
 }));
 
 describe("<LoginForm />", () => {
-    let mockLoginMutation;
     let mockDispatch;
 
     beforeEach(() => {
         mockDispatch = vi.fn();
         (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
 
-        mockLoginMutation = {
-            mutate: vi.fn(),
-            isError: false,
-        };
-        (useLoginMutation as jest.Mock).mockReturnValue(mockLoginMutation);
-
         window.virtool = {
             b2c: {
                 use: false,
             },
         };
     });
 
+    afterEach(() => nock.cleanAll());
+
     it("should call mutate() with correct values when submitted", async () => {
-        renderWithProviders(<LoginForm error="" />);
+        const username = "test_Username";
+        const password = "Password";
+
+        const scope = nock("http://localhost")
+            .post("/api/login", { username: username, password: password, remember: true })
+            .reply(200);
+
+        renderWithProviders(<LoginForm />);
 
         const usernameField = screen.getByLabelText("Username");
-        await userEvent.type(usernameField, "test_Username");
-        expect(usernameField).toHaveValue("test_Username");
+        await userEvent.type(usernameField, username);
+        expect(usernameField).toHaveValue(username);
 
         const passwordField = screen.getByLabelText("Password");
-        await userEvent.type(passwordField, "Password");
-        expect(passwordField).toHaveValue("Password");
+        await userEvent.type(passwordField, password);
+        expect(passwordField).toHaveValue(password);
 
         expect(screen.getByLabelText("Remember Me")).toHaveAttribute("data-state", "unchecked");
         await userEvent.click(screen.getByLabelText("Remember Me"));
         expect(screen.getByLabelText("Remember Me")).toHaveAttribute("data-state", "checked");
 
         await userEvent.click(screen.getByRole("button", { name: "Login" }));
-        await waitFor(() =>
-            expect(mockLoginMutation.mutate).toHaveBeenCalledWith(
-                { username: "test_Username", password: "Password", remember: true },
-                expect.any(Object),
-            ),
-        );
+
+        scope.isDone();
     });
 
     it("should display error message on login failure", async () => {
-        mockLoginMutation.isError = true;
+        const username = "test_Username";
+        const password = "Password";
+        const errorMessage = "An error occurred during login";
 
-        renderWithProviders(<LoginForm error="Invalid username or password" />);
+        const scope = nock("http://localhost")
+            .post("/api/login", { username: username, password: password, remember: false })
+            .reply(400);
```

Comment:
I would emulate returning the error message via the API. Makes for a less brittle test, (won't break if the hardcoded error messages changes) and tests the functionality we care more about

Reply:
So I did a bit of testing and was able to access the error message no problem at `error.response.body.message` with the following .reply configuration `.reply(400, { message: "test Error" });` 

While doing it I noticed a couple problems with your mocks that are probably what is giving you trouble. Will add those as additional comments for clarity sake

Reply:
Yes, those changes fixed the problem. Thank you!